js/style/parse_color.js

```diff
@@ -22,9 +22,14 @@ function parseColor(input) {
 
     // Color string
     } else if (typeof input === 'string') {
-        var output = colorDowngrade(parseCSSColor(input));
-        colorCache[input] = output;
-        return output;
+        var parsedColor = parseCSSColor(input);
+        if (parsedColor !== null) {
```

Comment:
Thanks for the style advise. Agreed, much better that way. I've updated the commit.